build.gradle.kts

```diff
@@ -12,7 +12,7 @@ allprojects {
 
 java {
     toolchain {
-        languageVersion.set(JavaLanguageVersion.of(16))
+        languageVersion.set(JavaLanguageVersion.of(17))
```

Comment:
Geyser-ViaProxy can target Java 17 without the whole project doing so - see how Geyser-Fabric does it: <https://github.com/GeyserMC/Geyser/blob/master/bootstrap/fabric/build.gradle.kts#L8-L11>